### ISSUE TITLE
" if event["type"] == "terminated":" error whenever I run bot.

### ISSUE THREAD

**Issue by ItsHypers** [ShailChoksi/lichess-bot#311]
Hello!

I'm having an issue, where whenever i run the bot, it gives the go ahead that's it up although crashes after around 20 seconds with this error:
![image](https://user-images.githubusercontent.com/59265736/114223997-68460580-9968-11eb-9376-006009808541.png)

I've seen people posting about this on here but no answers were given on their ticket so i made a new one!

**Comment by MarkZH** [comment]
Could you add the line `print(event)` below `event = control_queue.get()` on line 81 and report the output when the error occurs? Something like this:
```py
event = control_queue.get()
print(event) # <-- added line
if event["type"] == "terminated":
    break
```

**Comment by WhChess** [comment]
Same thing happened to me, and I did what you said, the result is:

```{'error': 'Missing scope'}
Traceback (most recent call last):
  File "lichess-bot.py", line 449, in <module>
    start(li, user_profile, engine_factory, CONFIG, logging_level, args.logfile)
  File "lichess-bot.py", line 131, in start
    if event["type"] == "terminated":
KeyError: 'type'```

**Comment by MarkZH** [comment]
@WhChess  Thanks for helping to debug this.

After googling "lichess missing scope", I found [this post](https://lichess.org/forum/lichess-feedback/upgrading-to-bot-account-error-missing-scope) and [this other post](https://lichess.org/forum/team-lichess-bots/missing-scope-unable-to-upgrade-to-bot-account). It may be that your API token was generated without the needed permissions to use the lichess APIs.

Try these steps:

1. Sign into the lichess.org website with your bot's account.
2. Go to the account profile.
3. Click the gear icon to edit profile.
4. Click "API Access tokens" (on the left on desktop, at the top on mobile)
5. Look at the API access token permissions and see what permissions it has.

For my bot's token, the list of permissions reads: "Read incoming challenges, Create, accept, decline challenges, Play games with the bot API".

The solution to this problem might be to delete your bot's current API access token and generating a new one, making sure to select all the needed permissions for the new token.

**Comment by WhChess** [comment]
Hello again, thank you for your help. I followed the steps you mentioned, over and over again. Problems arose again, it kept giving the same error. However, when I opened all the boxes except "Board Api" in the API token, the problem disappeared. I couldn't understand why it was like this, but my problem was solved as I explained. Thanks for your help. I wrote this message for those who experience this error like me.

**Comment by MarkZH** [comment]
Good to hear! It's odd that you had to select so many API scopes. I just generated a new token that only had the "Play games with the bot API" scope and it works for me.

Anyway, I'll add some new code to try to give a more useful error message in the future.

**Comment by WhChess** [comment]
" It's odd that you had to select so many API scopes."

Hi, I've been researching for a few hours as to why this is happening. I deleted all the old tokens, and created a new one, restarted the computer and started the bot again. And it worked as you said, maybe it's the problem with connecting the bot to lichess because I had set my first token only as "Play games with the bot API" and there was an error as same ; anyway maybe handling errors can really be the solution for this problem. Thank you for helping the issue, I hope it gets resolved.
